### ISSUE TITLE
Per-project/configuration build system default options file

### ISSUE THREAD

**Issue by ghost** [build2/build2#100]
Currently it is possible only system `/etc/build2/b.options` and user `.build2/b.options` config file. I think per project config (and may be per build config) will also be useful.

Related issue #98

**Comment by boris-kolpackov** [comment]
Yes, we thought about this and that's the semantics we have for `bpkg` and `bdep`. However, the build system case is tricky since one can be building multiple targets from different project and/or configurations in the same invocation. One idea we had is to start from the innermost common directory that encompasses all the targets specified on the command line but this makes the whole thing quite difficult to reason about (especially when we take into account forwarded configurations, imports, etc) and will complicate the implementation a lot (now we need to do quite a bit of work and then go back and (re-)parse options).

Also, it would be helpful if you described your specific use-case where you needed this semantics. That is, which build system option did you want to specify on a per-project/configuration basis?
